pkg/importer/dataStream_test.go

```diff
@@ -7,416 +7,404 @@ import (
-}
+	table.DescribeTable("can build selector", func(image, ep string, wantErr bool) {
+		if ep == "" {
+			ep = ts.URL
+		}
+		By(fmt.Sprintf("Creating new datastream for %s", ep+"/"+image))
+		ds, err := NewDataStream(ep+"/"+image, "", "")
+		if !wantErr {
+			Expect(err).NotTo(HaveOccurred())
+			err = ds.dataStreamSelector()
+			Expect(err).NotTo(HaveOccurred())
+		} else {
+			Expect(err).To(HaveOccurred())
+		}
+	},
+		table.Entry("success building selector for valid endpoint and file", "tinyCore.iso", "", false),
+		table.Entry("fail trying to build selector for invalid http endpoint", "", "http://www.google.com", true),
+		table.Entry("fail trying to build invalid selector", "tinyCore.iso", "fake://somefakefile", true),
+	)
+
+	table.DescribeTable("can construct readers", func(outfile, filename string, numRdrs int, wantErr bool) {
+		By(fmt.Sprintf("Creating new fileserver for %s and file %s", filepath.Dir(filename), filepath.Base(filename)))
+		tempTestServer := createTestServer(filepath.Dir(filename))
+		By(fmt.Sprintf("Creating new datastream to %s", tempTestServer.URL+"/"+filepath.Base(filename)))
+		ds, err := NewDataStream(tempTestServer.URL+"/"+filepath.Base(filename), "", "")
+		defer func() {
+			tempTestServer.Close()
+		}()
+
+		if !wantErr {
+			Expect(err).NotTo(HaveOccurred())
+			for _, r := range ds.Readers {
+				fmt.Fprintf(GinkgoWriter, "INFO: Reader type: %d\n", r.rdrType)
+			}
+			Expect(len(ds.Readers)).To(Equal(numRdrs))
+		} else {
+			Expect(err).To(HaveOccurred())
+		}
+	},
+		table.Entry("successfully construct a xz reader", "tinyCore.iso.xz", testfiles[".xz"], 5, false),              // [http, multi-r, xz, multi-r]
+		table.Entry("successfully construct a gz reader", "tinyCore.iso.gz", testfiles[".gz"], 5, false),              // [http, multi-r, gz, multi-r]
+		table.Entry("successfully construct qcow2 reader", "", filepath.Join(imageDir, "cirros-qcow2.img"), 2, false), // [http, multi-r]
+		table.Entry("successfully construct .iso reader", "", filepath.Join(imageDir, "tinyCore.iso"), 3, false),      // [http, multi-r]
+		table.Entry("fail constructing reader for invalid file path", "", filepath.Join(imageDir, "tinyCorebad.iso"), 0, true),
+	)
+})
+
+var _ = Describe("Copy", func() {
+	var ts *httptest.Server
+
+	BeforeEach(func() {
+		By("[BeforeEach] Creating test server")
+		ts = createTestServer(imageDir)
+	})
 
-func replaceQEMUOperations(replacement image.QEMUOperations, f func()) {
-	orig := qemuOperations
-	if replacement != nil {
-		qemuOperations = replacement
-		defer func() { qemuOperations = orig }()
+	AfterEach(func() {
+		By("[AfterEach] closing test server")
+		ts.Close()
+	})
+
+	table.DescribeTable("Image, with import source should", func(dest, endpt string, qemuOperations image.QEMUOperations, wantErr bool) {
+		By("Configuring end point")
+		if !strings.HasPrefix(endpt, "http") {
+			endpt = ts.URL + "/" + endpt
+		}
+		By(fmt.Sprintf("end point is %s", endpt))
+		defer os.RemoveAll(dest)
```

Comment:
so this is equivalent to `rm -r /tmp/cdi-test-copy` for the 1st and 2nd test cases, and `rm -r ./cirros-qcow2.raw` for the 3rd, 4th, and 5th test cases. Unless the rm -r target is a a dir I think it's clearer to use `os.Remove`.

Reply:
I guess since the `os.RemoveAll` err is ignored it's ok to call this even when `CopyImage` fails and there is no destination file...

Reply:
I copied this code verbatim from the original file, didn't inspect it too much, but I don't see a reason to use tmp if not needed (might not even have tmp in certain environments). Also switched to os.Remove